File: lessons/compiled_lang/cs/src/i20.subroutine.cs
using System;

class I20Subroutine
{
    static int pond = 500; // never mutated - Fish() only touches its own local copy
    static int captured = 0;

    static void Fish()
    {
        int pond = 500; // shadows the static field for the rest of this method
        pond -= 150;
        captured += 150;
    }

    static void Main()
    {
        Console.WriteLine($"We have {pond} in this pond.");

        Fish();
        Console.WriteLine($"Fishing from a local pond... We now have {pond} in the main pond.");

        Fish();
        Console.WriteLine($"Fishing from a local pond... We now have {pond} in the main pond.");

        Fish();
        Console.WriteLine($"Fishing from a local pond... We now have {pond} in the main pond.");

        Console.WriteLine($"We now have a total of {captured} fish captured");
    }
}

File: lessons/compiled_lang/rust/src/i20.subroutine.rs
const POND: i32 = 500; // never mutated - fish() only touches its own local copy

static mut CAPTURED: i32 = 0;

fn fish() {
    let local_pond = POND - 150; // never touches POND, and never leaves this scope
    let _ = local_pond;
    unsafe {
        CAPTURED += 150;
    }
}

fn main() {
    println!("We have {POND} in this pond.");

    fish();
    println!("Fishing from a local pond... We now have {POND} in the main pond.");

    fish();
    println!("Fishing from a local pond... We now have {POND} in the main pond.");

    fish();
    println!("Fishing from a local pond... We now have {POND} in the main pond.");

    let captured = unsafe { CAPTURED };
    println!("We now have a total of {captured} fish captured");
}
